contracts/access/IAccessControlDefaultAdminRules.sol

```diff
@@ -12,62 +12,137 @@ import "./IAccessControl.sol";
+     * @dev Returns the delay between each {defaultAdmin} transfer.
      *
-     * Requirements:
+     * A scheduled delay change will take effect as soon as the schedule passes, returning the new delay.
+     */
+    function defaultAdminDelay() external view returns (uint48);
+
+    /**
+     * @dev Returns a tuple of `newDelay` and `schedule`.
+     * The former will be the new {defaultAdminDelay} after the latter passes.
      *
-     * - Only can be called by the current `DEFAULT_ADMIN_ROLE` holder.
+     * A zero value in both indicates there's no pending delay change.
      *
-     * Emits a {DefaultAdminRoleChangeStarted}.
+     * NOTE: A zero value only for `newDelay` means that the next {defaultAdminDelay} will be zero.
      */
-    function beginDefaultAdminTransfer(address newAdmin) external;
+    function pendingDefaultAdminDelay() external view returns (uint48 newDelay, uint48 schedule);
 
     /**
-     * @dev Completes a `DEFAULT_ADMIN_ROLE` transfer.
+     * @dev Starts a {defaultAdmin} transfer by setting {pendingDefaultAdmin} with a {defaultAdminDelay} schedule.
      *
      * Requirements:
      *
-     * - Caller should be the pending default admin.
+     * - Only can be called by the current {defaultAdmin}.
+     *
+     * Emits a DefaultAdminRoleChangeStarted event.
+     */
+    function beginDefaultAdminTransfer(address newAdmin) external;
+
+    /**
+     * @dev Completes a {defaultAdmin} transfer.
+     *
      * - `DEFAULT_ADMIN_ROLE` should be granted to the caller.
      * - `DEFAULT_ADMIN_ROLE` should be revoked from the previous holder.
+     *
+     * Requirements:
+     *
+     * - Only can be called by the {pendingDefaultAdmin}'s `newAdmin` and if its `scheduled` has passed.
      */
     function acceptDefaultAdminTransfer() external;
 
     /**
-     * @dev Cancels a `DEFAULT_ADMIN_ROLE` transfer.
+     * @dev Cancels a {defaultAdmin} transfer if there was a {pendingDefaultAdmin}.
+     *
+     * A {pendingDefaultAdmin} not yet accepted can also be cancelled with this function.
      *
      * Requirements:
      *
-     * - Can be called even after the timer has passed.
-     * - Can only be called by the current `DEFAULT_ADMIN_ROLE` holder.
+     * - Only can be called by the current {defaultAdmin}.
+     *
+     * May emit a DefaultAdminTransferCanceled event.
      */
     function cancelDefaultAdminTransfer() external;
+
+    /**
+     * @dev Begins a {defaultAdminDelay} change by scheduling the change in a way in which the
+     * current delay is still guaranteed to be respected.
+     *
+     * The {pendingDefaultAdminDelay}'s schedule is defined such that `(its schedule + a delayed default admin transfer)`
+     * takes at least the current {defaultAdminDelay}, following that:
+     *
+     * - The schedule is `block.timestamp + (current delay - new delay)` if the delay is reduced.
+     * - The schedule is `block.timestamp + {defaultAdminDelayIncreaseWait}` if the delay is increased.
```

Comment:
you need to explain the rational. 

Is it that `current delay` might be too short, which would not allow the admin to cancel if the new delay is too long ?

Reply:
I improved the way is described, but I moved the rationale to `defaultAdminDelayIncreaseWait`.

```
     * With the purpose of avoiding doubly delayed transfers (waiting the new delay to start another delayed transfer),
     * an increased delay can relax the effect schedule, however, when the original delay is low, waiting only the old
     * one can pose a risk of accidentally locking the contract (i.e. increasing from 1 second to 10 years).
     *
     * For both avoiding double-delayed transfers or fast-effect increases, this function can be configured for a custom
     * {defaultAdminDelay} increase scheduling.
     *
```

Do you think this is clear?

Reply:
Maybe if the new delay is longer than the current delay, but smaller than the "i don't know what its called delay that is 5 days by default" ... then we can just use do accelerate the change

Reply:
Are you proposing a `Math.min(defaultAdminDelayIncreaseWait(), _currentDefaultAdminDelay)` instead? I definitely would like to do that but it carries the same problem as in a delay change from 1 second to 10 years.

I agree with your comments, so I'm open to ideas.